### DISCUSSION TITLE
Backquote bug

### DISCUSSION THREAD

**Discussion by DanielAjoy** [jrincayc/ucblogo-code#111] [General]
I'm reading the UCBLogo user manual:

Backquotes can be nested.  Substitution is done only for commas at the same depth as the backquote in which they are found:

```
SHOW `[a `[b ,[1+2] ,[foo ,[1+3] d] e] f]
[a ` [b , [1+2] , [foo 4 d] e] f]
```

It seems to me that only the inner most comma was resolved, which was not at the same depth as the backquotes.

**Comment by brianharvey** [comment]
The language may be unclear, but this is the correct behavior.  After the `[a `[b we're two levels deep in backquotes.  The ,[1+2] is just one level of comma, same with the ,[foo ...].  But the ,[1+3] is two commas deep, matching the two backquotes deep.  (Good timing, though -- I'm just implementing backquote for Snap!.)

**Reply by DanielAjoy** [reply]
Is the single backquote in the result after the "a" supposed to be there?

**Reply by brianharvey** [reply]
Yup!  From the Scheme manual:

<img width="605" alt="Screen Shot 2021-07-25 at 11 25 39 PM" src="https://user-images.githubusercontent.com/3885074/126942746-69176328-9fc8-4eba-ac61-30263c685439.png">
